File: Linear Data Structures/Arrays/Array Types Code Implementation/Two-Dimension Arrays/C/2D.c
#include <stdio.h>

int main() {
    int rows, cols;
    printf("Enter the number of rows: ");
    scanf("%d", &rows);
    printf("Enter the number of columns: ");
    scanf("%d", &cols);

    int arr[rows][cols];
    printf("Enter elements in the %dx%d matrix: \n", rows, cols);
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &arr[i][j]);
        }
    }

    printf("2D Array (Matrix): \n");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", arr[i][j]);
        }
        printf("\n");
    }

    return 0;
}

/* 
Sample Output:
Input:
Enter the number of rows: 2
Enter the number of columns: 3
Enter elements in the 2x3 matrix: 
1 2 3
4 5 6

Output:
2D Array (Matrix): 
1 2 3 
4 5 6 
*/

File: Linear Data Structures/Arrays/Array Types Code Implementation/Two-Dimension Arrays/Python/2D.py
# Getting user input
rows = int(input("Enter the number of rows: "))
cols = int(input("Enter the number of columns: "))

arr = []
print(f"Enter elements in the {rows}x{cols} matrix:")
for i in range(rows):
    row = list(map(int, input().split()))
    arr.append(row)

print("2D Array (Matrix):")
for row in arr:
    print(" ".join(map(str, row)))

# Sample Output:
# Input:
# Enter the number of rows: 2
# Enter the number of columns: 3
# Enter elements in the 2x3 matrix:
# 1 2 3
# 4 5 6
#
# Output:
# 2D Array (Matrix):
# 1 2 3
# 4 5 6
